### DISCUSSION TITLE
Drag of all composent one by one

### DISCUSSION THREAD

**Discussion by VictorRenkin** [peterdsharpe/AeroSandbox#125] [General]
Hi,
I'm participating in the AIAA competition where we need to create a glider. My idea was to decompose the drag analysis into 4 parts: Vortex drag, Profile drag, Interference drag, and finally, Protrusion drag. I want to analyze the drag for take-off, landing, and in sink, where the propeller is down, but I haven't found if you cover this kind of configuration. Moreover, we need to have the drag for each component, but I didn't see that either. Can you suggest how I might use your code?

**Comment by benberry3d** [comment]
If using AeroBuildup for the analysis each wing and fuselage component gives it's separate profile drag contributions, and then the total airplane induced and profile is also given:

```
aero = asb.AeroBuildup(
    airplane=airplane,
    op_point=op_point,
).run()

induced_drag_total = aero["D_induced"]
profile_drag_total = aero["D_profile"]
profile_drag_wing = aero["wing_aero_components"][0].D
profile_drag_tail = aero["wing_aero_components"][1].D
profile_drag_fuselage = aero["fuselage_aero_components"][0].D
```

As far as I know interference drag and protrusion (or excrescence) drag is not modelled, and you would have to multiply your own factors to profile drag or however you want to account for those drags.  I would actually love to see an excrescence drag factor added as an input option to the AeroBuildup or Airplane objects to multiply by total profile drag, to help with book-keeping.  Right now I just multiply a factor by aero["D_profile"] and then add that to total drag for performance calculations.

**Reply by VictorRenkin** [reply]
Thanks a lot. Do you also know if you can have the lift distribution for the cord, in your example is it just the profile drag.

**Reply by benberry3d** [reply]
Could you clarify what you mean by " lift distribution for the cord"?  Perhaps explaining what you need it for.  The short answer for any lift distribution will be "no" from AeroBuildup.  The Vortex Lattice Method would output spanwise (or chordwise) force distributions:
[Vortex Lattice Tutorial](https://github.com/peterdsharpe/AeroSandbox/blob/master/tutorial/06%20-%20Aerodynamics/01%20-%20AeroSandbox%203D%20Aerodynamics%20Tools/01%20-%20Vortex%20Lattice%20Method/01%20-%20Vortex%20Lattice%20Method.ipynb)
[Another VLM example](https://github.com/peterdsharpe/AeroSandbox/blob/master/tutorial/07%20-%20Structures/03%20-%20Coupled%20Aerostructures/01%20-%20Steady%20Wing%20Aerostructural%20Analysis.ipynb)
In the second link, see the parts of the code that say "vlm.forces_geometry" which is an example of how you can extract the output forces on the wing.
